crates/tools/js/benchmark/index.js

```diff
@@ -0,0 +1,216 @@
+      Math.round(100 * (timeMs / 1000)) / 100
+    } seconds with ${failures.length} failures.`
+  );
+
+  return {
+    name,
+    result: {
+      timeMs,
+      failures,
+    },
+  };
+}
+
+async function loadScenario(path) {
+  const result = {
+    requests: [],
+  };
+  let i = 0;
+  for await (const line of readFile(path)) {
+    const parsed = JSON.parse(line);
+    if (i === 0) {
+      result.config = preprocessConfig(parsed);
+    } else {
+      result.requests.push(parsed);
+    }
+    i += 1;
+  }
+  return result;
+}
+
+function preprocessConfig(config) {
+  // From https://stackoverflow.com/a/59771233
+  const camelize = (obj) =>
+    _.transform(obj, (acc, value, key, target) => {
+      const camelKey = _.isArray(target) ? key : _.camelCase(key);
+
+      acc[camelKey] = _.isObject(value) ? camelize(value) : value;
+    });
+  config = camelize(config);
+
+  // EDR serializes None as null to json, but Hardhat expects it to be undefined
+  const removeNull = (obj) =>
+    _.transform(obj, (acc, value, key) => {
+      if (_.isObject(value)) {
+        acc[key] = removeNull(value);
+      } else if (!_.isNull(value)) {
+        acc[key] = value;
+      }
+    });
+  config = removeNull(config);
+
+  config.providerConfig.initialDate = new Date(
+    config.providerConfig.initialDate.secsSinceEpoch * 1000
+  );
+
+  config.providerConfig.hardfork = normalizeHardfork(
+    config.providerConfig.hardfork
+  );
+
+  // "accounts" in EDR are "genesisAccounts" in Hardhat
+  if (Object.keys(config.providerConfig.genesisAccounts).length !== 0) {
+    throw new Error("Genesis accounts are not supported");
+  }
+  config.providerConfig.genesisAccounts = config.providerConfig.accounts.map(
+    ({ balance, secretKey }) => {
+      return { balance, privateKey: secretKey };
+    }
+  );
+  delete config.providerConfig.accounts;
+
+  config.providerConfig.automine = config.providerConfig.mining.autoMine;
+  config.providerConfig.mempoolOrder =
+    config.providerConfig.mining.memPool.order.toLowerCase();
+  config.providerConfig.intervalMining =
+    config.providerConfig.mining.interval ?? 0;
+  delete config.providerConfig.mining;
+
+  config.providerConfig.throwOnCallFailures =
+    config.providerConfig.bailOnCallFailure;
+  delete config.providerConfig.bailOnCallFailure;
+  config.providerConfig.throwOnTransactionFailures =
+    config.providerConfig.bailOnTransactionFailure;
+  delete config.providerConfig.bailOnTransactionFailure;
+
+  let chains = new Map();
+  for (let key of Object.keys(config.providerConfig.chains)) {
+    const hardforkHistory = new Map();
+    const hardforks = config.providerConfig.chains[key].hardforks;
+    for (let [blockNumber, hardfork] of hardforks) {
+      hardforkHistory.set(normalizeHardfork(hardfork), blockNumber);
+    }
+    chains.set(Number(key), { hardforkHistory });
+  }
+  config.providerConfig.chains = chains;
+
+  if (!_.isUndefined(config.providerConfig.fork)) {
```

Comment:
The `_` is very confusing to me. Can't we just import `isUndefined` from lodash?

Reply:
It's standard to use `lodash` like this